src/Development/IDE/Core/OfInterest.hs

```diff
@@ -79,4 +81,9 @@ modifyFilesOfInterest state f = do
     OfInterestVar var <- getIdeGlobalState state
     files <- modifyVar var $ pure . dupe . f
     logDebug (ideLogger state) $ "Set files of interest to: " <> T.pack (show $ HashSet.toList files)
-    void $ shakeRun state []
+    void $ shakeRun state [kick]
+
+kick :: Action ()
+kick = do
+    files <- getFilesOfInterest
+    void $ uses TypeCheck $ HashSet.toList files
```

Comment:
The reason this was outside the core was because it's wrong. In DAML we have a much better kick operation that takes care of garbage collection etc which we one day planned to move over. This was meant to be a temporary hack. But that doesn't mean it's in the wrong place now, but something to think about for the future.

Reply:
Do these improvements still apply? If so, I can add a TODO comment if you or @cocreature can be more precise about the nature of the improvements.